src/Signature/S3SignatureV4.php

```diff
@@ -41,6 +41,11 @@ public function presign(
                 'X-Amz-Content-Sha256',
                 $this->getPresignedPayload($request)
             );
+        } else {
+            $request = $request->withHeader(
```

Comment:
I believe so. The test must be discarding the supplied value and using what's returned by `getPresignedPayload` for both scenarios.